js/validation.js

```diff
@@ -0,0 +1,56 @@
+
+const validateTypeofKey = function (key, newValue, type) {
+  if (typeof newValue !== type) {
+    throw new Error(`${key} has to be ${type}`)
+  }
+}
+
+const allKeys = ['id', 'title', 'ingredients', 'instructions', 'total_calories', 'number_of_portions', 'calories_per_portion']
+const isValidKey = function (key) {
+  return allKeys.includes(key)
+}
+
+/** Returns true for a valid recipe key-value pair, throws exception otherwise */
+const validateRecipeKey = function (key, newValue) {
+  switch (key) {
+    case 'title':
+    case 'instructions':
+      validateTypeofKey(key, newValue, 'string')
+      break
+    case 'total_calories':
+    case 'calories_per_portion':
+    case 'number_of_portions':
+    case 'id':
+      validateTypeofKey(key, newValue, 'number')
+      break
+    case 'ingredients':
+      if (!(Array.isArray(newValue))) {
```

Comment:
Is the extra paren in the middle required?

Reply:
No, thanks